source/euler/indicator.h

```diff
@@ -228,29 +274,64 @@ namespace ryujin
       const auto m_j = view.momentum(U_j);
       const auto f_j = view.f(U_j);
 
-      left += (eta_j * rho_j_inverse - eta_i * rho_i_inverse) * (m_j * c_ij);
-      for (unsigned int k = 0; k < problem_dimension; ++k)
-        right[k] += (f_j[k] - f_i[k]) * c_ij;
+      const auto entropy_flux =
+          (eta_j * rho_j_inverse - eta_i * rho_i_inverse) * (m_j * c_ij);
+
+      if (parameters.indicator_strategy() == IndicatorStrategy::evc_fullsplit) {
+        /* Entropy viscosity commutator with aggressive denominator split: */
+
+        left_absolute += std::abs(entropy_flux);
+        for (unsigned int k = 0; k < problem_dimension; ++k) {
+          const auto component = d_eta_i[k] * (f_j[k] - f_i[k]) * c_ij;
+          right_value += component;
+          right_absolute += std::abs(component);
+        }
+
+      } else {
+        /* Entropy viscosity commutator with conservative denominator split: */
+
+        left += entropy_flux;
+        for (unsigned int k = 0; k < problem_dimension; ++k) {
+          const auto component = (f_j[k] - f_i[k]) * c_ij;
+          right[k] += component;
+        }
+      }
     }
 
 
     template <int dim, typename Number>
     DEAL_II_ALWAYS_INLINE inline Number
     Indicator<dim, Number>::alpha(const Number hd_i) const
     {
-      using ScalarNumber = typename get_value_type<Number>::type;
+      if (parameters.indicator_strategy() == IndicatorStrategy::galerkin)
+        return Number(0.);
 
-      Number numerator = left;
-      Number denominator = std::abs(left);
-      for (unsigned int k = 0; k < problem_dimension; ++k) {
-        numerator -= d_eta_i[k] * right[k];
-        denominator += std::abs(d_eta_i[k] * right[k]);
-      }
+      if (parameters.indicator_strategy() == IndicatorStrategy::evc_fullsplit) {
+        /* Entropy viscosity commutator with aggressive denominator split: */
 
-      const auto quotient =
-          std::abs(numerator) / (denominator + hd_i * std::abs(eta_i));
+        Number numerator = left - right_value;
+        Number denominator = left_absolute + right_absolute;
 
-      return std::min(Number(1.), parameters.evc_factor() * quotient);
+        const auto quotient =
+            std::abs(numerator) / (denominator + hd_i * std::abs(eta_i));
+
+        return std::min(Number(1.), parameters.evc_factor() * quotient);
+
+      } else {
+        /* Entropy viscosity commutator with conservative denominator split: */
+
+        Number numerator = left;
+        Number denominator = std::abs(left);
+        for (unsigned int k = 0; k < problem_dimension; ++k) {
+          numerator -= d_eta_i[k] * right[k];
+          denominator += std::abs(d_eta_i[k] * right[k]);
+        }
+
+        const auto quotient =
+            std::abs(numerator) / (denominator + hd_i * std::abs(eta_i));
+
+        return std::min(Number(1.), parameters.evc_factor() * quotient);
+      }
```

Comment:
@kronbichler The runtime penalty for this if statement is measurable. But right now I would like to avoid exposing this if statement as a constexpr in the `HyperbolicModule`. It seems to be too PDE dependent.

I evaluated some alternatives as well such as simply computing both branches at the same time, but that was slower.